### ISSUE TITLE
feature request: "get current location" button as with android

### ISSUE THREAD

**Issue by naturalistcharlie** [inaturalist/INaturalistIOS#292]
I now have a work phone that is an Android and so I have been using the Android version of the app sometimes too. While the android app has its upsides and downsides it's a million times better than it was last time I used it. One of the things I really like is the 'get current location' button, it helps when the initial location marked came up a bit off, and it would also probably help me with my own weird mapping bug that no one else can replicate that comes up when I don't get a good GPS reading. Any chance we could get that on iOS too?

**Comment by loarie** [comment]
Hi Charlie,

The equivalent to 'get current location' does exist on iOS:

1. On the Edit Obs Detail screen tap on the location row in the form to
launch the map
2. On the Edit Location page click the little pointer icon in the lower
left below the map

That should center the map on and set the coords to your current location

On Tue, Jun 21, 2016 at 8:22 PM, Charlie <notifications@github.com> wrote:

> I now have a work phone that is an Android and so I have been using the
> Android version of the app sometimes too. While the android app has its
> upsides and downsides it's a million times better than it was last time I
> used it. One of the things I really like is the 'get current location'
> button, it helps when the initial location marked came up a bit off, and it
> would also probably help me with my own weird mapping bug that no one else
> can replicate that comes up when I don't get a good GPS reading. Any chance
> we could get that on iOS too?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/inaturalist/INaturalistIOS/issues/292>, or mute the
> thread
> <https://github.com/notifications/unsubscribe/AAkgCiz8bcD1hMlfrakUJKgwf6ptHDZxks5qOI5agaJpZM4I7Usa>
> .
>



-- 
--------------------------------------------------
Scott R. Loarie, Ph.D.
Co-director, iNaturalist.org
California Academy of Sciences
55 Music Concourse Dr
San Francisco, CA 94118
--------------------------------------------------

**Comment by naturalistcharlie** [comment]
Thanks Scott! However that takes a lot longer because you have to save and
then re-edit the observation, right? In the Android version it's just one
easy tap within the create observation screen. Of course, as always, maybe
it's something only I want and no one else would use and thus not worth
creating. Mostly I am trying to find ways to deal with both my weird
mapping bug and with general app mapping accuracy QC and such.

 In any event last I checked editing location was totally broken on the app
and it *always* defaulted back to where the observation was initially taken
(or in the case of that bug only i seem to have, back to the wrong
location). I know others have noted that too, but i guess i haven't tried
it in the most recent version.

C

On Wed, Jun 22, 2016 at 8:58 PM, Scott Loarie <notifications@github.com>
wrote:

> Hi Charlie,
>
> The equivalent to 'get current location' does exist on iOS:
>
> 1. On the Edit Obs Detail screen tap on the location row in the form to
> launch the map
> 2. On the Edit Location page click the little pointer icon in the lower
> left below the map
>
> That should center the map on and set the coords to your current location
>
> On Tue, Jun 21, 2016 at 8:22 PM, Charlie <notifications@github.com> wrote:
>
> > I now have a work phone that is an Android and so I have been using the
> > Android version of the app sometimes too. While the android app has its
> > upsides and downsides it's a million times better than it was last time I
> > used it. One of the things I really like is the 'get current location'
> > button, it helps when the initial location marked came up a bit off, and
> it
> > would also probably help me with my own weird mapping bug that no one
> else
> > can replicate that comes up when I don't get a good GPS reading. Any
> chance
> > we could get that on iOS too?
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/inaturalist/INaturalistIOS/issues/292>, or mute the
> > thread
> > <
> https://github.com/notifications/unsubscribe/AAkgCiz8bcD1hMlfrakUJKgwf6ptHDZxks5qOI5agaJpZM4I7Usa
> >
> > .
> >
>
>
>
> --
> --------------------------------------------------
> Scott R. Loarie, Ph.D.
> Co-director, iNaturalist.org
> California Academy of Sciences
> 55 Music Concourse Dr
> San Francisco, CA 94118
> --------------------------------------------------
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/inaturalist/INaturalistIOS/issues/292#issuecomment-227921270>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AG1tk__epcR4wcYev9eNVJZXwhdH4yUjks5qOdpPgaJpZM4I7Usa>
> .
>



-- 
============================
Charlie Hohn
Montpelier, Vermont

**Comment by kueda** [comment]
I'm with Charlie on this one. My main beef with the map button is that if you don't have network, you can't tell if it's working or not because it won't load any map tiles. By doing it on the obs edit form, you can show a spinner and then see the numbers populate and the accuracy go down. Feedback is a good thing.

**Comment by alexshepard** [comment]
@joellebel any thoughts on this?

**Comment by joellebel** [comment]
Hi there. Haven't forgot about this. Just thinking...

**Comment by joellebel** [comment]
I'm hesitant to do this but don't want to add more buttons to the map view. So you just wanna go with what we have on Android? Tapping the location would bring up an action sheet with the options "Get current location" and "Edit location". I'm only not too nuts about this because it's one more step to go through before getting to the map, but can see how this interaction is handy.

@alexshepard let me know if you need a mock up or I'm leaving out any states and/ or scenarios.

**Comment by alexshepard** [comment]
Unfortunately adding button(s) to the map view wouldn't solve it anyways, based on Ken-ichi's comment.

I don't really like the action sheet either. I'd prefer an action on the Edit screen.

**Comment by joellebel** [comment]
@alexshepard For now though, is the action sheet the quickest solution implementation-wise?

**Comment by alexshepard** [comment]
Yeah, this or another standard UI implementation like a simple button or something will be easiest.

**Comment by joellebel** [comment]
ok. Will mock up so we can compare.

**Comment by joellebel** [comment]
Haven't forgotten about this. Will get on it soon...

**Comment by joellebel** [comment]
Here's 4 variations... This would only show up if there has been no location recorded. 

![image](https://cloud.githubusercontent.com/assets/11527994/17843767/c6a57ee8-67e6-11e6-9ed3-48ff6d53e2fb.png)


I prefer A and B. A more than B. I made B in case it's not clear that you can still tap anywhere else in the cell to get to the map. 

I made C in case long phrases was a concern with other languages.

**Comment by kueda** [comment]
Direction B gets my vote.

**Comment by alexshepard** [comment]
I prefer option B because I feel it's more verbose and clear. 

I am concerned about how longer languages will fit in Option B. Option C helps there, but the button titles in C don't clearly indicate indicate to me the actions to be taken.

Just so it doesn't get lost, we'll also need feedback indicators while the location fetch is happening and getting more precise.

**Comment by alexshepard** [comment]
For the "choose location on the map" action, I prefer the pin+map icon in Option D over the circle+plus icon used in the other options. It seems to more clearly indicate what tapping the button will do.

Can we consider replacing the `Add Location` button title with something like `Add From Map`? `Use Current Location` also adds a location, so `Add Location` isn't a distinct alternative choice.

**Comment by joellebel** [comment]
I'm cool with Add from Map...

ha. I knew folks would like the pin+map icon. I like it to, but it would need refactoring then added to the icon font. For now let's just stick with what's in our icon arsenal. I also replaced the icons with their solid counterparts because they're easier to read at that size.

Below is what happens on iphone 6 and smaller screens. @alexshepard can we make the "Add from map" gracefully fall to the second line. This would also help for longer languages. 

![image](https://cloud.githubusercontent.com/assets/11527994/17948961/14c3fadc-6a08-11e6-975a-3b32857894eb.png)




> Just so it doesn't get lost, we'll also need feedback indicators while the location fetch is happening and getting more precise.

What about replacing the "Use Current..." icon with the native spinner while the app is looking for the location.

**Comment by alexshepard** [comment]
That's fine
